core/src/test/java/org/apache/brooklyn/core/mgmt/persist/ListeningObjectStore.java

```diff
@@ -209,6 +213,20 @@ public void put(String val) {
             delegate.put(val);
         }
         @Override
+        public void put(ByteSource val) {
+            if (writesFailSilently)
+                return;
+
+            for (ObjectStoreTransactionListener listener: listeners) {
+                try {
+                    listener.recordDataOut("writing (binary) "+path, Streams.readFullyAndClose(val.openStream()).length);
```

Comment:
Is it really worth the extra read and close just to be able to write the length here?